### PR TITLE
Update due-diligence-guidelines.md to be explicit about dependencies

### PR DESCRIPTION
**Pull request by chira001**
Be explicit about collecting dependencies and listing them

### CODE REVIEW

**Comment by chira001** [comment]
@lizrice @quinton-hoole @erinaboyd - making a short addition to the DD guidelines to make documenting dependencies an explicit requirement

File: process/due-diligence-guidelines.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ the detail where necessary.
 * What are the most important holes? No High-Availability? No flow control? Inadequate integration points?
 * Code quality.  Does it look good, bad or mediocre to you (based on a spot review).  How thorough are the code reviews? Substance over form.
   Are there explicit coding guidelines for the project?
-* Dependencies.  What external dependencies exist, do they seem justified?
+* Dependencies.  What external dependencies exist, do they seem justified?  Note: all core dependencies should be listed in the document along with the details of relevant repos
```
Comment (dims) [inline process/due-diligence-guidelines.md:85]: "core" implies "non-optional"?

Comment (chira001) [reply process/due-diligence-guidelines.md:85]: yes - that was the intent - happy to change it if there is better wording.

Comment (dims) [reply process/due-diligence-guidelines.md:85]: "non-optional" seems better otherwise folks will debate what "core" is/means :)

**Comment by dims** [comment]
@chira001 do we want to add another list item about cross checking licenses for all dependencies and make a list of things that do are not white listed here? https://github.com/cncf/foundation/blob/master/allowed-third-party-license-policy.md

( this is a slightly different problem then the original above )

File: process/due-diligence-guidelines.md (hunk 1 above)
Comment (lizrice) [reply process/due-diligence-guidelines.md:85]: I don't know how to get the terminology right here but I think we need to distinguish between significant functional dependences, and programming language / package dependencies. We don't want to be reviewing giant lists of, say, Go packages or Node dependencies, but we do want to know when a project assumes the existence of a significant external component.

**Comment by chira001** [comment]
> @chira001 do we want to add another list item about cross checking licenses for all dependencies and make a list of things that do are not white listed here? https://github.com/cncf/foundation/blob/master/allowed-third-party-license-policy.md
> 
> ( this is a slightly different problem then the original above )

I think this is already covered by "Does it have a license? Which one? Does it have a CLA or DCO? Are the licenses of it's dependencies compatible with their usage and CNCF policies? CNCF staff will handle the full legal due diligence."  and the graduation policy already specifies that the project needs to comply with the CNCF IP Policy, but you are right - it would be helpful to include a reference in the guidelines.

**Comment by erinaboyd** [comment]
@chira001 @lizrice really what we are trying to get at though is "does it have a non-open source 3rd party license?"
Maybe that would be easier listing than all the possible licenses

**Comment by dims** [comment]
@erinaboyd we already have a list :) https://github.com/cncf/foundation/blob/master/allowed-third-party-license-policy.md

**Pull request merged**